backend/src/main/java/com/empire/World.java

```diff
@@ -761,33 +761,50 @@ void setGothiSpellStatus() {
-				tivar.deluge = true;
-			} else if (tivar.deluge) {
-				tivar.deluge = Math.random() < 0.5;
-				if (!tivar.deluge) notifyAllPlayers("Deluge Ends", "The magical deluge drowning the land has finally ceased.");
+				if (tivar.deluge == 0) notifyAllPlayers("The Deluge Begins", "The necessary number of gothi of Syrjen have agreed to call forth the Deluge. Terrible floods make all land regions treacherous (if navigable by fleets), and more crops are drowned in the muddy waters every week.");
+				tivar.deluge++;
+			} else if (tivar.deluge > 0) {
+				if (Math.random() > Constants.tivarSpellContinueChance) {
+					tivar.deluge = 0;
+					notifyAllPlayers("Deluge Ends", "The magical deluge drowning the land has finally ceased.");
+				} else {
+					tivar.deluge++;
+				}
 			}
 			if (passedSpells.contains("Lyskr")) {
-				if (!tivar.veil) notifyAllPlayers("The Veil Falls", "The necessary number of gothi of Lyskr have agreed to call forth the Veil. Heavy fog chokes out the sun and reduces visibility severely. Barred from the sunlight, crops will be lost every week until the fog lifts.");
-				tivar.veil = true;
-			} else if (tivar.veil) {
-				tivar.veil = Math.random() < 0.5;
-				if (!tivar.veil) notifyAllPlayers("Veil Ends", "The magical fog blanketing the land has finally lifted.");
+				if (tivar.veil == 0) notifyAllPlayers("The Veil Falls", "The necessary number of gothi of Lyskr have agreed to call forth the Veil. Heavy fog chokes out the sun and reduces visibility severely. Barred from the sunlight, crops will be lost every week until the fog lifts.");
+				tivar.veil++;
+			} else if (tivar.veil > 0) {
+				if (Math.random() > Constants.tivarSpellContinueChance) {
+					tivar.veil = 0;
+					notifyAllPlayers("Veil Ends", "The magical fog blanketing the land has finally lifted.");
+				} else {
+					tivar.veil++;
+				}
 			}
 			if (passedSpells.contains("Alyrja")) {
-				if (!tivar.warwinds) notifyAllPlayers("The Warwinds Howl", "The necessary number of gothi of Alyrja have agreed to call forth the Warwinds. Titanic waves make seagoing treacherous, and powerful winds blow all vessels at sea off course. The temperature plummets, killing crops every week until the storm ceases.");
-				tivar.warwinds = true;
-			} else if (tivar.warwinds) {
-				tivar.warwinds = Math.random() < 0.5;
-				if (!tivar.warwinds) notifyAllPlayers("Warwinds End", "The magical storms devastating the land have finally calmed.");
+				if (tivar.warwinds == 0) notifyAllPlayers("The Warwinds Howl", "The necessary number of gothi of Alyrja have agreed to call forth the Warwinds. Titanic waves make seagoing treacherous, and powerful winds blow all vessels at sea off course. The temperature plummets, killing crops every week until the storm ceases.");
```

Comment:
This also seems misleading about the grace period.  I may have actually missed some earlier, you might just want to check all of the descriptions if this comment is warranted.

Reply:
Yep, I missed these four. Fixed.